synapse/handlers/_base.py

```diff
@@ -133,10 +133,9 @@ def handle_new_client_event(self, event, context, extra_destinations=[],
         for k, s in context.current_state.items():
             try:
                 if k[0] == EventTypes.Member:
-                    if s.content["membership"] == Membership.JOIN:
```

Comment:
We want to always send membership events for a user to their homeserver, not just if its a join event.